x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_signal.ts

```diff
@@ -5,35 +5,41 @@
  */
 
 import { RulesSchema } from '../../../../common/detection_engine/schemas/response/rules_schema';
-import { INTERNAL_IDENTIFIER } from '../../../../common/constants';
 import { SignalSourceHit, Signal, Ancestor } from './types';
 
-export const buildAncestor = (doc: SignalSourceHit, rule: Partial<RulesSchema>): Ancestor => {
-  const existingSignal = doc._source.signal?.parent;
-  if (existingSignal != null) {
+/**
+ * Takes a parent signal or event document and extracts the information needed for the corresponding entry in the child
+ * signal's `signal.parents` array.
+ * @param doc The parent signal or event
+ */
+export const buildParent = (doc: SignalSourceHit): Ancestor => {
+  if (doc._source.signal != null) {
     return {
-      rule: rule.id != null ? rule.id : '',
+      rule: doc._source.signal.rule.id,
       id: doc._id,
       type: 'signal',
       index: doc._index,
-      depth: existingSignal.depth + 1,
+      // We first look for signal.depth and use that if it exists. If it doesn't exist, this should be a pre-7.10 signal
+      // and should have signal.parent.depth instead. signal.parent.depth in this case is treated as equivalent to signal.depth.
+      depth: doc._source.signal.depth ?? doc._source.signal?.parent?.depth ?? 1,
     };
   } else {
     return {
-      rule: rule.id != null ? rule.id : '',
       id: doc._id,
       type: 'event',
       index: doc._index,
-      depth: 1,
+      depth: 0,
     };
   }
 };
 
-export const buildAncestorsSignal = (
-  doc: SignalSourceHit,
-  rule: Partial<RulesSchema>
-): Signal['ancestors'] => {
-  const newAncestor = buildAncestor(doc, rule);
+/**
+ * Takes a parent signal or event document with N ancestors and adds the parent document to the ancestry array,
+ * creating an array of N+1 ancestors.
+ * @param doc The parent signal/event for which to extend the ancestry.
+ */
+export const buildAncestors = (doc: SignalSourceHit): Signal['ancestors'] => {
```

Comment:
Nit: Are `Signal['ancestors']` and `Ancestor[]` equivalent? If they are, should we use `Ancestor[]` for consistency, just so as a dev quickly looking through here it's clear `buildParent` and `buildAncestors` are returning the same type just one is array and the other isn't?